### PR TITLE
Fix access log

### LINKED ISSUE #14121: access.log stays empty
- Gitea version (or commit ref): 1.14.0+dev-449-ge0c753e77 (latest docker image (non-rootless))
- Git version: 
- Operating system:
  
  
  
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [x ] No
- Log gist:

## Description

I did activate the access log in the config using 

DISABLE_ROUTER_LOG=false
ENABLE_ACCESS_LOG=true
ACCESS=file

access.log file appears in the expected location but stays empty. If you tail it you can see that on incoming requests emtpy lines are added. 

gitea.log contains error messages: 
2020/12/23 14:49:49 ...outers/routes/chi.go:71:1() [E] Could not set up macaron access logger: template: log:1:6: executing "log" at <.Ctx.RemoteAddr>: can't evaluate field Ctx in type routes.routerLoggerOptions

### PR DESCRIPTION
**Pull request by lunny**
Fix #14121, #14478.

The `AccessLog` middleware has to be after `Contexter` or `APIContexter` so that we can get `LoginUserName` if possible.
And also there is a **BREAK** change that it removed internal API access log.

### CODE REVIEW

File: modules/auth/sso/oauth2.go [hunk 1]
```diff
@@ -12,6 +12,7 @@ import (
 
 	"code.gitea.io/gitea/models"
 	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/middlewares"
```
Comment (techknowlogick) [inline modules/auth/sso/oauth2.go:15]: `middlewares` should be `middleware` as we use singular for others

Comment (lunny) [reply modules/auth/sso/oauth2.go:15]: #14480

**Comment by 6543** [comment]
@lunny can you use middleware instead of middleware**s**?

**Comment by lunny** [comment]
> @lunny can you use middleware instead of middleware**s**?

see https://github.com/go-gitea/gitea/pull/14475#discussion_r564977446

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
:point_up: since it's breaking ...

**Review by zeripath — APPROVED** [review]

**Pull request merged**